h5py/_hl/base.py

```diff
@@ -416,6 +416,10 @@ def items(self):
             """ Get a view object on member items """
             return ItemsViewHDF5(self)
 
+    def _ipython_key_completions_(self):
+        """ Custom tab completions for __getitem__ in IPython >=5.0. """
+        return sorted(self.keys())
```

Comment:
Empirically, these seem to come back sorted already, but I couldn't find a guarantee of that fact. Regardless, we need to force the list in py3, so this seems like a reasonable way to do that either way.

Reply:
It depends on whether you specify `track_order=True` when creating the file/group - if so, they're in creation order. Otherwise, they should be in name order.

Reply:
Ah, ok. Does it seem reasonable to guarantee that these come back sorted? I'm not sure what IPython's display logic is for these. Do they get sorted anyway downstream?

Reply:
Looking at IPython's completion code, I think it sorts results anyway. But it's fairly convoluted, and I don't think it's an API guarantee, so sorting them seems a reasonable thing to do.